rustls-mio/tests/common/mod.rs

```diff
@@ -192,13 +192,26 @@ pub fn openssl_find() -> String {
         return format!("{}/bin/openssl", dir);
     }
 
-    // We need a homebrew openssl, because OSX comes with
-    // 0.9.8y or something equally ancient!
+    // We may need a homebrew openssl, because older versions of OSX come with
+    // 0.9.8y or something equally ancient.
     if cfg!(target_os = "macos") {
-        if Path::new("/usr/local/opt/openssl@1.1/bin/openssl").is_file() {
-            return "/usr/local/opt/openssl@1.1/bin/openssl".to_string();
-        } else {
-            return "/usr/local/opt/openssl/bin/openssl".to_string();
+        match process::Command::new("brew")
+            .args(&["--prefix", "openssl"])
+            .output() {
+            Ok(output) => {
+                let dir = str::from_utf8(&*output.stdout).unwrap().trim();
+                return format!("{}/bin/openssl", dir);
+            },
+            Err(_) => {
+                if Path::new("/usr/local/opt/openssl@1.1/bin/openssl").is_file() {
```

Comment:
Maybe extract the possible paths as a const slice of strs?

Reply:
Sure thing. I try not to make stuff like this too concise, so it's easy to see what happens when something goes wrong, but I don't feel strongly about this case.